src/cli/serve/serve.js

```diff
@@ -112,12 +112,11 @@ export function applyConfigOverrides(rawConfig, opts, extraCliOptions) {
     if (opts.serverless) {
       setServerlessKibanaDevServiceAccountIfPossible(get, set, opts);
 
-      // Load mock identity provider plugin and configure realm if supported (ES only supports SAML when run with SSL)
+      // Configure realm if supported (ES only supports SAML when run with SSL)
       if (opts.ssl && MOCK_IDP_PLUGIN_SUPPORTED) {
         // Ensure the plugin is loaded in dynamically to exclude from production build
         // eslint-disable-next-line import/no-dynamic-require
         const { MOCK_IDP_REALM_NAME } = require(MOCK_IDP_PLUGIN_PATH);
-        const pluginPath = resolve(require.resolve(MOCK_IDP_PLUGIN_PATH), '..');
```

Comment:
__note:__ It turned out plugins in `packages` are discovered automatically already.